### PR TITLE
Add CLI config and remove conditional compile flags

### PR DESCRIPTION
**Pull request by rishflab**
The hardcoded configuration was replaced with CLI
configuration options. CLI based config was chosen
over a config file as it does not access and clutter
the user's file system. By CLI options depend on whether
the program is run in Alice or Bob mode.

The tor conditional compile flags were removed by
extracting transport creation to the main statement. A tor
transport is created if Alice specifies a tor port using the CLI.

### CODE REVIEW

File: swap/src/main.rs [hunk 1]
```diff
@@ -30,95 +37,97 @@ mod trace;
...
-pub const ADDR: &str = "127.0.0.1";
-pub const BITCOIND_JSON_RPC_URL: &str = "http://127.0.0.1:8332";
-
-#[cfg(feature = "tor")]
-pub const TOR_PORT: u16 = PORT + 1;
-
 #[tokio::main]
 async fn main() -> Result<()> {
-    let opt = Options::from_args();
+    let opt: Options = Options::from_args();
```
Comment (luckysori) [inline swap/src/main.rs:42]: :wrench: Looks unnecessary.

File: swap/src/cli.rs [hunk 2]
```diff
@@ -1,19 +1,35 @@
+use libp2p::core::Multiaddr;
+use url::Url;
+
 #[derive(structopt::StructOpt, Debug)]
-pub struct Options {
-    /// Run the swap as Alice.
-    #[structopt(long = "as-alice")]
-    pub as_alice: bool,
+pub enum Options {
```
Comment (luckysori) [inline swap/src/cli.rs:5]: :wrench: Looks like some docstrings are missing and/or outdated here.

File: swap/src/main.rs [hunk 3]
```diff
@@ -30,95 +37,97 @@ mod trace;
...
-        info!("running swap node as Alice ...");
-
-        if opt.piconeros.is_some() || opt.satoshis.is_some() {
-            bail!("Alice cannot set the amount to swap via the cli");
+    match opt {
+        Options::Alice {
+            bitcoind_url: url,
+            listen_addr,
+            tor_port,
+            ..
```
Comment (luckysori) [inline swap/src/main.rs:51]: :question: Aren't you matching against all fields?

File: swap/src/main.rs [hunk 4]
```diff
@@ -30,95 +37,97 @@ mod trace;
...
+                .expect("failed to create bitcoin wallet");
+
+            let redeem = bitcoin_wallet
+                .new_address()
+                .await
+                .expect("failed to get new redeem address");
+            let punish = bitcoin_wallet
+                .new_address()
+                .await
+                .expect("failed to get new punish address");
```
Comment (luckysori) [inline swap/src/main.rs:88]: Once you rebase address generation should be inside of the `{alice, bob}::swap` functions. Just an FYI.

File: swap/src/main.rs [hunk 5]
```diff
@@ -30,95 +37,97 @@ mod trace;
...
+            listen_addr,
+            tor_port,
+            ..
+        } => {
+            info!("running swap node as Alice ...");
+
+            let behaviour = Alice::default();
+            let local_key_pair = behaviour.identity();
+
+            let (listen_addr, _ac, transport) = match tor_port {
```
Comment (luckysori) [inline swap/src/main.rs:58]: :question: Is it only Alice that can run her binary through `tor`? @bonomat is this how it's meant to be?

**Review by luckysori — APPROVED** [review]
LGTM!

File: swap/src/main.rs (hunk 5 above)
Comment (bonomat) [reply swap/src/main.rs:58]: If Bob always dials Alice and Alice is behind Tor then both parties need Tor.

Differently said: 
* `Tor` --> _dials_ --> `No-Tor`: ✔️
* `Tor` --> _dials_ --> `Tor`: ✔️
* `No-Tor` --> _dials_ --> `No-Tor`: ✔️
* `No-Tor` --> _dials_ --> `Tor`: x

Bob does not need to add a tor service as Alice won't dial Bob.

File: swap/src/main.rs (hunk 1 above)
Comment (rishflab) [reply swap/src/main.rs:42]: far out this one keeps getting me, doesn't IDE can't handle it for some reason

File: swap/src/main.rs (hunk 5 above)
Comment (luckysori) [reply swap/src/main.rs:58]: Thanks for the explanation! We therefore must support Bob starting the binary with Tor. Is that currently possible, @rishflab?

File: swap/src/main.rs (hunk 1 above)
Comment (bonomat) [reply swap/src/main.rs:42]: `swap` is completely broken in my IDE :(

File: swap/src/main.rs (hunk 3 above)
Comment (rishflab) [reply swap/src/main.rs:51]: yep :)

File: swap/src/main.rs (hunk 5 above)
Comment (rishflab) [reply swap/src/main.rs:58]: Yep, fixed

**Pull request merged**